packages/ses/src/lockdown.js

```diff
@@ -100,10 +100,32 @@ const safeHarden = makeHardener();
 // only ever need to be called once and that simplifying lockdown will improve
 // the quality of audits.
 
-const assertDirectEvalAvailable = () => {
-  let allowed = false;
+const probeHostEvaluators = () => {
+  let functionAllowed;
   try {
-    allowed = FERAL_FUNCTION(
+    functionAllowed = FERAL_FUNCTION('return true')();
+  } catch (_error) {
+    // We reach here if the Function() constructor is outright forbidden by a
+    // strict Content Security Policy (containing either a `default-src` or a
+    // `script-src` directive), not been implemented in the host, or the host
+    // is configured to throw an exception instead of `new Function`.
```

Comment:
Shouldn't `'no-eval'` also disallow these?

Reply:
This is where we check what the capabilities of the host are. 
`no-eval` should have an effect after this check. 

If the above is not correct, we have to look into running this earlier or changing the logic of asserting on the results of `probeHostEvaluators` a bit.